pkgs/pkgs-lib/formats/hocon/default.nix

```diff
@@ -97,36 +97,52 @@ in
               optional = value.optional or false;
               _type = "substitution";
             };
-      };
 
+        types = {
```

Comment:
@Stunkymonkey do you have any thoughts about using `format.lib.types` as the place for these? Would it be better to use something like `format.subtypes` and reserve the `lib` attrset for functions only, or do we view `lib` as an equivalent to the global `lib` with equivalent namespacing? (`format.types` is probably too close to `format.type` and will likely cause typos with possibly confusing error messages)

There's seemingly no other formats exposing these sort of relevant types at the moment, so this PR will probably set the precedent.